src/main/java/redis/clients/jedis/BinaryJedisPubSub.java

```diff
@@ -115,6 +129,9 @@ private void process(Client client) {
         subscribedChannels = ((Long) reply.get(2)).intValue();
         final byte[] bpattern = (byte[]) reply.get(1);
         onPUnsubscribe(bpattern, subscribedChannels);
+      } else if (Arrays.equals(PONG.raw, resp)) {
+        final byte[] bpattern = (byte[]) reply.get(1);
```

Comment:
What does this `reply.get(1)` return when `ping()` is called without an argument?

Reply:
Good question!  It actually returns an empty string as bytes.  However, we currently return `null` in `JedisPubSub` if no argument is provided:
```
else if (Arrays.equals(PONG.raw, resp)) {
        final byte[] bpattern = (byte[]) reply.get(1);
        final String strpattern = (bpattern == null) ? null : SafeEncoder.encode(bpattern);
        onPong(strpattern);
}
```

Perhaps we should return null in `BinaryJedisPubSub` as well instead of an empty string?

Reply:
@sazzad16 do we need to protect this use case?

Reply:
@gkorland Yes. PING with message replies PONG with message here. Check `onPong` in `binaryPubSubChannelWithPingPongWithArgument` method.